internal/server/storage/dbstorage.go

```diff
@@ -0,0 +1,170 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+
+	"github.com/npavlov/go-metrics-service/internal/domain"
+	"github.com/npavlov/go-metrics-service/internal/model"
+)
+
+type DBStorage struct {
+	db *gorm.DB
```

Comment:
[CHANGE REQUEST] It would be better if you used `sqlc`
https://sqlc.dev/

Reply:
Using sqlc but facing a lot of issues, need your consultation